### ISSUE TITLE
Question about Expert.

### ISSUE THREAD

**Issue by BertyWooster** [Rudrabha/Wav2Lip#111]
Sorry, can you help me please to understand, what did you exactly mean: " Our
expert lip-sync discriminator is about 91% accurate on the LRS2 test
set." ?

Is my understanding correct?
In case of sync sample accurate exert gives score more than 0.5. In the other case expert gives score less  than 0.5. We count such successful cases and devide them to all number of LRS test samples. We should get percentage about 91.

**Comment by prajwalkr** [comment]
Hello

We mean the ``off-sync detection accuracy", which is mentioned in Table 3 of the paper (Section 4.5). By off-sync detection, we mean the ability of the model to give a score of less than 0.5 for off-sync audio-video pairs. It is computed by forming random off-sync audio-video pairs from the LRS test samples.

**Comment by BertyWooster** [comment]
First, thank you for response and great work!
But I have one more question. Is this evaluation metric takes into account the ability of model to determine sync pairs?
Will the Expert gives scores more then 0.5 for sync pairs with similar quality (about 91%)?

**Comment by prajwalkr** [comment]
The provided model actually gives a higher accuracy at detecting in-sync pairs. In the paper, we specifically wanted to highlight off-sync detection accuracy.

**Comment by BertyWooster** [comment]
Thank you very much! :)

**Comment by zhanghm1995** [comment]
> Hello
> 
> We mean the `off-sync detection accuracy`, which is mentioned in Table 3 of the paper (Section 4.5). By off-sync detection, we mean the ability of the model to give a score of less than 0.5 for off-sync audio-video pairs. It is computed by forming random off-sync audio-video pairs from the LRS test samples.

Hello, @prajwalkr , I want to know how did you construct the off-sync audio-video pairs in LRS2 dataset? You know the number of off-sync audio-video pairs could be very large since the random audio and video meet the off-sync requirements. Therefore, I wonder was there some rules to construct the off-sync audio-video pairs?

**Comment by Mayur28** [comment]
Hi @BertyWooster , @prajwalkr ,

I just wanted to clarify which score are we using to determine sync (or off-sync)? I see that many of the previous comments make mention of a "score", and if the score is <1, it is off-sync and in-sync otherwise. I just wanted to find out how to compute the score please? Is the score taken to be the output of `evaluations/SyncNetInstance_calc_scores`? And are we taking the confidence score as the sync score, or the min distance, or something else?

Thanks!

**Comment by BertyWooster** [comment]
Hi @Mayur28 
 
[SyncNet_color](https://github.com/Rudrabha/Wav2Lip/blob/master/models/syncnet.py#:~:text=class%20SyncNet_color(nn.Module)%3A) get **audio_sequences** and **face_sequences** to return couple of vectors  - **audio_embedding**, **video_embedding**

* These vectors represent compressed information about video and audio input

* There are nn.ReLU() activation at the end of SyncNet_color, so we can only get not negative dot product between these embedding vectors

* so we get values from 0 to 1 as a measure of sync of pair (audio and video sample, 5 frames duration)
https://github.com/Rudrabha/Wav2Lip/blob/master/color_syncnet_train.py#:~:text=logloss%20%3D%20nn.BCELoss()-,def,-cosine_loss(a%2C%20v

0 - not sync, 1 - sync

after that, we can use BCE loss to our network learn classification sync/not sync

**Comment by Mayur28** [comment]
Hi @BertyWooster,

Thanks for the detailed explanation, really appreciate it!

Just to clarify once more, we are considering the sync score to be the output of the cosine similarity [here](https://github.com/Rudrabha/Wav2Lip/blob/b9759a3467cb1b7519f1a3b91f5a84cb4bc1ae4a/color_syncnet_train.py#L135), as this would be in accordance with your explanation in that a cosine sim. < 0.5 would indicate off-sync and cosine sim > 0.5 would indicate in-sync?

Thanks!

**Comment by BertyWooster** [comment]
Yes, I think so. But this model can perform not so well to score unseen data. I think we have to train it on new data distribution to get the correct sync score.

**Comment by BertyWooster** [comment]
@Mayur28

**Comment by Mayur28** [comment]
Hi @BertyWooster ,

Thanks for clarifying, your explanation certainly makes sense and has pointed me in the right direction and I will now look into it further, thanks again!
